contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,200 @@
+
+    [source](https://hackr.io/blog/why-should-you-learn-angular-in-2018)
+
+### Benefit: Angular provides a declarative user interface.
+
+Angular uses HTML to define the app’s user interface. HTML is a declarative language which is more intuitive than using imperative code to define the UI. An Imperative approach requires you to tell the compiler how you want the program to run by focussing on the logic implementation. Declarative code, on the other hand enables you to describe the logic without worrying about the implementation. How does it help? You do not need to invest time in dealing with how your program runs and deciding what to load first.
+
+Lets see a working example. 
+
+With jQuery you might write the following code to define a textbox that takes numeric values and doesn’t allow alphanumeric input or input that is beyond of a defined range.
+
+```
+$("#textbox").numericTextBox({
+    value: 10,
+    min: -10,
+    max: 100,
+    step: 0.75,
+    format: "n",
+    decimals: 3
+});
+```
+On the other hand, using versions Angular 2 and above, your code would look something like this:
+
+```
+<input numerictextbox k-min="-10" k-max="100" k-step="0.75"
+  k-format="n" k-decimals="3"/>
+```
+Instead of worrying about how to transform the element, you can declare what you want the user to see.
+Rather than spending time thinking on how the program should flow, you can define what you want and Angular will take care of the dependencies.
+
+[source](https://www.telerik.com/blogs/three-ds-of-web-development-1-declarative-vs-imperative)
+
+### Benefit: Simplified MVC pattern and reduced coding.
+
+Angular framework is embedded with the MVC (Model-View-Controller) software architectural setup. The HTML, is what is presented to the user (along with some CSS for layout). This represents the View component. Next to it, we have the `component.ts` files. This is the controller. Essentially, it can choose which data to push to our view (`.html`). 
+Lastly, we have the model. In Angular, the model will mostly be our services, which we can access through our controller. Services are a way to retrieve, update and process data to and from the backend.
+![MVC Pattern](https://www.grazitti.com/assets/2018/06/reasons-5.jpg)
+
+Most frameworks require developers to split the application into multiple MVC components. After that, the developer has to write code to put them together. Angular, however, strings it together automatically. As explained in this [article](https://blog.angular-university.io/why-angular-angular-vs-jquery-a-beginner-friendly-explanation-on-the-advantages-of-angular-and-mvc/) on the official website, everything happens under the hood. Angular ensures Separation of Concerns and automatic synchronization between the components. That saves developers time.
```

Comment:
> strings it together automatically

Not sure if there is a better term to describe what you want to convey but I think it can be better rephrased.